### PR TITLE
#16017184 Create a landing page and basic UI Setup

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### What does this PR do?
This PR creates a landing page for the project and also adds the basic UI set up on the home page after the user has moved away from the landing page.
#### Background context
The landing page is the first page a user will see when they visit the application. The user can then click anywhere on the landing page to see the content on the home page. The home page has a signin and signup button that reveal the login and register components in a modal. The home page also shows recent article and most popular articles.
#### What are the relevant pivotal tracker stories?
[#16017184](https://www.pivotaltracker.com/story/show/160171840)

### CODE REVIEW

File: src/components/Landing/Landing.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        height: auto;
+        padding: 6px 7px;
+        top: 33%;
+        left: 0px;
+        padding-bottom: 50px;
+    }
+    .Headline button{
+        width: 144px;
+    }
+}
```
Comment (ssewilliam) [inline src/components/Landing/Landing.css]: @AGMETEOR 
This file is missing a new line

File: src/tests/Home.test.js [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+  });
+
+  it("should render the popular articles component", () => {
+    expect(wrapper.find(Popular)).toHaveLength(1);
+  });
+
+  it("should change the show props to false", () => {
+
+      wrapper.instance().hideLandingPage()
+    //   expect(wrapper.instance().hideLandingPage()).toBe(false);
```
Comment (ssewilliam) [inline src/tests/Home.test.js]: @AGMETEOR 
This looks like commented code

**Review by ssewilliam — COMMENTED** [review]
@AGMETEOR, Thanks for the contribution
- There are just a few changes needed
- Also, the commit message did not follow guidelines
 <i>see here: </i> https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices

File: .eslintrc.json [hunk 3]
```diff
@@ -31,7 +41,7 @@
     "react/no-undef": 0,
     "react/no-unknown-property": 1,
     "react/prefer-es6-class": 1,
-    "react/prop-types": 1,
+    "react/prop-types": 0,
```
Comment (geofrocker) [inline .eslintrc.json]: why did you deactivate this rule?

File: package.json [hunk 4]
```diff
@@ -98,7 +105,7 @@
     "extends": "react-app"
   },
   "devDependencies": {
-    "eslint": "^5.4.0",
+    "eslint": "^4.19.1",
```
Comment (geofrocker) [inline package.json]: Why are you downgrading eslint?

File: src/UI/Backdrop/Backdrop.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.Backdrop {
+    width: 100%;
+    height: 100%;
+    position: fixed;
+    z-index: 888;
+    left: 0;
+    top: 0;
+    background-color: rgba(0, 0, 0, 0.7);
+}
```
Comment (geofrocker) [inline src/UI/Backdrop/Backdrop.css]: missing EOF

File: src/UI/Modal/Modal.css [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    box-sizing: border-box;
+    transition: all 0.5s ease-out;
+}
+
+@media(min-width: 600px){
+    .Modal{
+        width: 500px;
+        left: calc(50% - 250px);
+    }
+}
```
Comment (geofrocker) [inline src/UI/Modal/Modal.css]: Missing EOF

File: src/UI/Modal/Modal.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import classes from './Modal.css';
+import Backdrop from '../Backdrop/Backdrop';
+import Wrapper from '../../hoc/Wrapper/Wrapper';
+
+const modal = props => (
+  <Wrapper>
+    <div
+      className={classes.Modal}
+      style={{
```
Comment (geofrocker) [inline src/UI/Modal/Modal.js]: Remove in-line css

File: src/components/Footer/Footer.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+const footer = () => (
+  <footer
+    style={{
```
Comment (geofrocker) [inline src/components/Footer/Footer.js]: same here

File: src/components/Landing/Landing.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    <div
+      className={divClasses}
+      style={{
+        transform: !props.show ? 'translateX(-100vw)' : null,
+      }}
+      onClick={props.closeLanding}
+    >
+      <div className={classes.Headline}>
+        <div className="col-sm-6">
+          <h4 className="h1">Author's Haven</h4>
```
Comment (geofrocker) [inline src/components/Landing/Landing.js]: Why should an h4 have a className of h1

File: src/store/index.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import {
```
Comment (geofrocker) [inline src/store/index.js]: This file should be inside reducers

File: src/tests/toggleLandingPageReducer.test.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import reducer from "../store/reducers/toggleLandingPageReducer";
+import * as actionTypes from "../store/actions/actionTypes";
```
Comment (geofrocker) [inline src/tests/toggleLandingPageReducer.test.js]: Remove wild card imports

**Review by geofrocker — CHANGES_REQUESTED** [review]

**Comment by geofrocker** [comment]
Resolve merge conflicts

File: src/CSS/index.css [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  
+  .authorTag {
+    font-size: 12px;
+    color: #838080;
+  }
+  
+  .articleImg {
+    width: 350px;
+    height: 200px;
+  }
```
Comment (geofrocker) [inline src/CSS/index.css]: missing EOF

**Review by geofrocker — CHANGES_REQUESTED** [review]

**Review by abulojoshua1 — APPROVED** [review]

**Review by lytes20 — APPROVED** [review]

File: .eslintrc.json [hunk 13]
```diff
@@ -3,7 +3,18 @@
...
+    "arrow-parens":0,
+    "arrow-body-style":0,
+    "no-named-as-default":0,
+    "jsx-a11y/anchor-is-valid":["enabled", "never"],
+    "react/button-has-type":["enabled", "never"],
+    "jsx-a11y/no-static-element-interactions":["enabled", "never"],
+    "jsx-a11y/click-events-have-key-events":["enabled", "never"],
+    "import/no-named-as-default":["enabled", "never"],
+    "react/no-unescaped-entities": ["enabled", "never"],
+    "react/destructuring-assignment": ["enabled", "never"],
```
Comment (geofrocker) [inline .eslintrc.json:17]: why did you disable this rule?

**Review by phillipseryazi — APPROVED** [review]

**Review by RutaleIvanPaul — COMMENTED** [review]
Could you setup a css folder in an appropriate location to hold the css for the entire app? Considering that you are creating the base UI.

File: .eslintrc.json (hunk 13 above)
Comment (AGMETEOR) [reply .eslintrc.json:17]: They were not very necessary.

Comment (geofrocker) [reply .eslintrc.json:17]: They don't have to be "very" necessary. Destructuring is a javascript best practice

Comment (AGMETEOR) [reply .eslintrc.json:17]: Alright. Let me work on it.

**Pull request merged**